monix-eval/shared/src/main/scala/monix/eval/TaskLocal.scala

```diff
@@ -140,11 +140,7 @@ final class TaskLocal[A] private (default: => A) {
     *        reset to the previous value
     */
   def bind[R](value: A)(task: Task[R]): Task[R] =
-    Task.suspend {
-      val saved = ref.value
-      ref.update(value)
-      task.doOnFinish(_ => restore(saved))
-    }
+    Task.suspend(bindL(Task.now(value))(task))
```

Comment:
If you suspend that value as I mentioned below, you don't need to suspend here.

Reply:
@alexandru  Understood, will take care of it later today.